pkg/cloud-init/cloud-init.go

```diff
@@ -168,14 +168,11 @@ func ResolveSecrets(source *v1.CloudInitNoCloudSource, namespace string, clients
 		return err
 	}
 
-	userDataBase64, ok := secret.Data["userdata"]
+	userData, ok := secret.Data["userdata"]
```

Comment:
Does the client also decode it back?

Reply:
yes. Our tests passed because we double-en and  double-decoded it ....